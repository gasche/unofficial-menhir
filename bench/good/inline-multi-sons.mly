%token A B EOF
%start<int> s
%%

(* We test the freshness of producer's name during inlining. 
   ioption and delimited both use 'x' in their definition.
*)
s: ioption(pair(ioption(A), delimited(A, B, A))) EOF { 0 }
