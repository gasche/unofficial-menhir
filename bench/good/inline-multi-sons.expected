Warning: you are using the standard library and/or the %inline keyword. We
recommend switching on --infer in order to avoid obscure type error messages.
%start s
%token EOF
%token B
%token A

%type <int> s
%%

s:
| EOF
    {let _1 =
  ( None )
in
 0 }
| A x000 = B A EOF
    {let _1 =
  let x00 = x000 in
  let x =
    let x0 = x00 in
    let y =
      let x = x0 in
      ( x )
    in
    let x =
      ( None )
    in
    ( (x, y) )
  in
  ( Some x )
in
 0 }
| x000 = A A x100 = B A EOF
    {let _1 =
  let x10 = x100 in
  let x00 = x000 in
  let x =
    let x1 = x10 in
    let x0 = x00 in
    let y =
      let x = x1 in
      ( x )
    in
    let x =
      let x = x0 in
      ( Some x )
    in
    ( (x, y) )
  in
  ( Some x )
in
 0 }

%%



